File: Section-12 Selection Sort/selection_sort.js
function swap(array,low,high){
    let temp = array[low];
    array[low] = array[high];
    array[high] = temp;
}


//O(n^2)
function selection_sort(array) {
    for (let i = 0; i < array.length; i++) {
        let index_of_min_value = i;
        for (let j = i+1; j < array.length; j++) {
            if (array[j]<array[index_of_min_value]){
                index_of_min_value = j;
            }
        }
        //swap
        swap(array,i,index_of_min_value);
    }
    return array;
}


console.log(selection_sort([3,2,1,8,9,0,5,6,2,4,7]))

File: Section-12 Selection Sort/selection_sort.py
#O(n^2)
def selection_sort(array):
    for i in range(len(array)):
        index_of_min_value = i
        for j in range(i+1,len(array)):
            if array[j]<array[index_of_min_value]:
                index_of_min_value = j

        #swap
        array[i],array[index_of_min_value] = array[index_of_min_value],array[i]
    return array

print(selection_sort([3,2,5,1,5,6,0,72,2]))